src/AppInstallerCLICore/VTSupport.cpp

```diff
@@ -164,4 +164,63 @@ namespace AppInstaller::CLI::VirtualTerminal
         const Sequence EraseLineBackward{ AICLI_VT_CSI "1K" };
         const Sequence EraseLineEntirely{ AICLI_VT_CSI "2K" };
     }
+
+    namespace Progress
+    {
+        ConstructedSequence Construct(State state, std::optional<uint32_t> percentage)
+        {
+            // See https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+
+            THROW_HR_IF(E_BOUNDS, percentage.has_value() && percentage > 100u);
+
+            // Workaround some quirks in the Windows Terminal implementation of the progress OSC sequence

```

Comment:
Y'all, we work at the same company. If I keep finding "quirk for Windows Terminal" comments I'm gonna have to glare at you from across the lunch room. 😁

Reply:
I will keep that in mind @DHowett , although this one is more of an external report through us, so maybe we can still be friends?

Reply:
I am to blame for writing that mean comment, sorry. 😁 I wanted to check whether this was a known issue/limitation (I think I saw an issue with some follow-up tasks related to the initial implementation in Windows Terminal) but forget it in the end.

I see you haven't yet created an issue report so let me do it.

Reply:
I created an issue for this in the Windows Terminal repo: https://github.com/microsoft/terminal/issues/9960

Reply:
Thanks a ton!

(sorry -- my joke probably came across very poorly over the medium of text. Thanks for being understanding! 😁)